### PR TITLE
task: update example go

### PR DESCRIPTION
**Pull request by gssbzn**
## Description

Update the example, Go keeps trying to also update this file so I want to commit the change t avoid doing it by accident

### CODE REVIEW

**Review by wtrocki — APPROVED** [review]

File: examples/go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module go.mongodb.org/atlas-sdk/v20240805004/examples
 
-go 1.21
+go 1.22.0
+
+toolchain go1.23.1
```
Comment (lantoli) [inline examples/go.mod:5]: what are the pros/cons of defining this toolchain as opposed to not define it?

File: examples/go.mod [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
 
 replace go.mongodb.org/atlas-sdk/v20240805004 => ../
```
Comment (lantoli) [inline examples/go.mod:7]: @gssbzn what are the pros/cons of defining this toolchain as opposed to not define it?

Comment (gssbzn) [reply examples/go.mod:7]: toolchain is suppose to be for local dev vs as a lib

Comment (wtrocki) [reply examples/go.mod:7]: Yes. We always need to keep go version 2 versions behind latest but tooling can be latest (and we should align our internal system go version with tooling version)

Comment (lantoli) [reply examples/go.mod:7]: but this is for examples, can it be used as a library or it should be fine just to use go 1.23.1 in examples' go.mod?

Comment (wtrocki) [reply examples/go.mod:7]: Does not make difference for me as you properly pointed out sdk use case does not apply for examples.

**Pull request merged**